lib/Driver/WebAssemblyToolChains.cpp

```diff
@@ -0,0 +1,253 @@
+//===---- WebAssemblyToolChains.cpp - Job invocations (WebAssembly-specific) ------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2021 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+#include "ToolChains.h"
+
+#include "swift/ABI/System.h"
+#include "swift/AST/DiagnosticsDriver.h"
+#include "swift/Basic/LLVM.h"
+#include "swift/Basic/Platform.h"
+#include "swift/Basic/Range.h"
+#include "swift/Config.h"
+#include "swift/Driver/Compilation.h"
+#include "swift/Driver/Driver.h"
+#include "swift/Driver/Job.h"
+#include "swift/Option/Options.h"
+#include "swift/Option/SanitizerOptions.h"
+#include "clang/Basic/Version.h"
+#include "clang/Driver/Util.h"
+#include "llvm/ADT/StringSwitch.h"
+#include "llvm/Option/Arg.h"
+#include "llvm/Option/ArgList.h"
+#include "llvm/ProfileData/InstrProf.h"
+#include "llvm/Support/FileSystem.h"
+#include "llvm/Support/Path.h"
+#include "llvm/Support/Process.h"
+#include "llvm/Support/Program.h"
+
+using namespace swift;
+using namespace swift::driver;
+using namespace llvm::opt;
+
+std::string toolchains::WebAssembly::sanitizerRuntimeLibName(StringRef Sanitizer,
+                                                         bool shared) const {
+  return (Twine("clang_rt.") + Sanitizer + "-" +
+          this->getTriple().getArchName() + ".lib")
+      .str();
+}
+
+ToolChain::InvocationInfo toolchains::WebAssembly::constructInvocation(
+    const AutolinkExtractJobAction &job, const JobContext &context) const {
+  assert(context.Output.getPrimaryOutputType() == file_types::TY_AutolinkFile);
+
+  InvocationInfo II{"swift-autolink-extract"};
+  ArgStringList &Arguments = II.Arguments;
+  II.allowsResponseFiles = true;
+
+  addPrimaryInputsOfType(Arguments, context.Inputs, context.Args,
+                         file_types::TY_Object);
+  addInputsOfType(Arguments, context.InputActions, file_types::TY_Object);
+
+  Arguments.push_back("-o");
+  Arguments.push_back(
+      context.Args.MakeArgString(context.Output.getPrimaryOutputFilename()));
+
+  return II;
+}
+
+ToolChain::InvocationInfo
+toolchains::WebAssembly::constructInvocation(const DynamicLinkJobAction &job,
+                                         const JobContext &context) const {
+  assert(context.Output.getPrimaryOutputType() == file_types::TY_Image &&
+         "Invalid linker output type.");
+
+  ArgStringList Arguments;
+
+  std::string Target = getTriple().str();
+  if (!Target.empty()) {
+    Arguments.push_back("-target");
+    Arguments.push_back(context.Args.MakeArgString(Target));
+  }
```

Comment:
`--target` option is added twice https://github.com/apple/swift/pull/39299/files#diff-678e83f2487547ddadcb72231e5d6a8dd296c0fb001889af4aba5271f23420abR161-R163

Reply:
Good catch, I cleaned that up.